### PR TITLE
feat(CB_GF): correct update rule and simulation unit test

### PR DESCRIPTION
**Pull request by olgavrou**
- allow multiple costs if graph feedback

### CODE REVIEW

File: vowpalwabbit/core/include/vw/core/reductions/cb/cb_explore_adf_common.h [hunk 1]
```diff
@@ -105,6 +105,7 @@ class cb_explore_adf_base
       const multi_ex& ec_seq, VW::io::logger& logger);
 
   ExploreType explore;
+  static bool allow_multiple_costs;
```
Comment (jackgerrits) [inline vowpalwabbit/core/include/vw/core/reductions/cb/cb_explore_adf_common.h:108]: Does this need to be static?

Comment (olgavrou) [reply vowpalwabbit/core/include/vw/core/reductions/cb/cb_explore_adf_common.h:108]: yes since it is used in predict/learn which are static

Comment (jackgerrits) [reply vowpalwabbit/core/include/vw/core/reductions/cb/cb_explore_adf_common.h:108]: If there are two workspace instances in the same process, one using cb with graph feedback and one not using it then this static would essentially break that scenario.

If this needs to be dynamic based on the setup then the variable and learn/predict should be non static

Comment (olgavrou) [reply vowpalwabbit/core/include/vw/core/reductions/cb/cb_explore_adf_common.h]: actually it doesn't need to be static and good point, updated

**Review by jackgerrits — APPROVED** [review]

**Pull request merged**